ui/raidboss/data/07-dt/raid/r4n.ts

```diff
@@ -1,14 +1,274 @@
+import Outputs from '../../../../../resources/outputs';
+import { Responses } from '../../../../../resources/responses';
 import ZoneId from '../../../../../resources/zone_id';
 import { RaidbossData } from '../../../../../types/data';
 import { TriggerSet } from '../../../../../types/trigger';
 
-export type Data = RaidbossData;
+const effectB9AMap = {
+  orangeDiamondFront: '2D3',
+  blueCircleBack: '2D4',
+} as const;
+
+type B9AMapKeys = keyof typeof effectB9AMap;
+type B9AMapValues = typeof effectB9AMap[B9AMapKeys];
+
+export interface Data extends RaidbossData {
+  expectedBlasts: 3 | 4 | 5;
+  storedBlasts: B9AMapValues[];
+}
+
+const b9aValueToNorthSouth = (
+  searchValue: B9AMapValues | undefined,
+): 'north' | 'south' | 'unknown' => {
+  if (searchValue === effectB9AMap.blueCircleBack) {
+    return 'north';
+  } else if (searchValue === effectB9AMap.orangeDiamondFront) {
+    return 'south';
+  }
+
+  return 'unknown';
+};
+
+const isEffectB9AValue = (value: string | undefined): value is B9AMapValues => {
+  if (value === undefined)
+    return false;
+  return Object.values<string>(effectB9AMap).includes(value);
+};
+
+// TODO: Map out MapEffect data if needed? Might be useful for prep for savage.
+// TODO: Better triggers for Sidewise Spark. Some sort of phase detection and collect setup is needed
+// as well as identifying the clones based on npc ID or something
+// TODO: Same thing for Bewitching Flight, collector for the loop to combine trigger with clone
+// TODO: Witch Hunt, determine starting point and figure out how to word the dodge
+
+// TODO: Might be able to use these to detect phase push, I didn't look into it very much
+const npcYellData = {
+  // Offsets: 456920,494045,510794
```

Comment:
This is direct output from the `auto-triggers` branch. `Offsets` are milliseconds since start of pull, e.g. the first time that `M-My body...` was detected was at 456920ms = 456.920 seconds = 7m 36.920s.

I find that offsets are less useful for fights with a phase change like this one, unfortunately, since they're meant for aggregate comparison across multiple pulls but phase change means that they can vary significantly.

Either way this is probably better discussed on #222